oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,92 @@ impl super::Node for WasmNode {
     }
 }
 
-#[test]
-fn wasm_invalid_args() {
-    use super::Node;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::node::Node;
     use crate::runtime::TEST_NODE_ID;
-    let configuration = crate::runtime::Configuration {
-        nodes: HashMap::new(),
-        entry_module: "test_module".to_string(),
-        entrypoint: "test_function".to_string(),
-    };
-    let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
-    let (_, reader_handle) =
-        runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
-
-    // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
-    // Minimal module:
-    // \0asm - magic
-    // 0x01 - version (in little-endian)
-    let module =
-        wasmi::Module::from_buffer(vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]).unwrap();
-
-    let mut node = WasmNode::new(
-        "test",
-        runtime_ref,
-        TEST_NODE_ID,
-        Arc::new(module),
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use std::env;
+    use wat::parse_file;
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
+        // Minimal module:
+        // \0asm - magic
+        // 0x01 - version (in little-endian)
```

Comment:
Not sure this comment belongs here, it was meant to explain where `vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]` comes from in the original example.

Reply:
I have moved it to `wasm_starting_module_without_content_fails` now.